frontend-ssr/pages/submissions.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Head from 'next/head';
+import Footer from '~/components/footer/Footer.component';
+import Header from '~/components/header/Header.component';
+
+import styles from '../styles/pages/default-page.module.scss';
+import style from './submissions.module.scss';
+import ExSubmission from '../components/exercise-submissions/ExerciseSubmission';
+import ExerciseService from '../services/ExerciseSubmissions.service';
+import { loadSubmissions } from '~/redux/exercise-submissions/exercise-submissions.actions';
+import SubmissionSkeleton from '~/components/exercise-submissions/ExerciseSubmissionSkeleton';
+import SweetAlertService from '../services/sweet-alert/SweetAlert.service';
+import store from '~/redux/store';
+
+interface State {
+  loading: boolean;
+}
+interface Props {
+  submissions: any;
+  dispatch: any;
+}
+
+class Submissions extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+    };
+  }
+
+  componentDidMount() {
+    let { submissions } = this.props;
+    const state = submissions;
+    function shouldFetchSubmissions() {
```

Comment:
This entire thing seems overly complicated. You have another inner function declared here, which is usually not a good practice, then you also call the store directly and get it's state, but don't do anything with it 🤔

It should be much simpler: if we don't have submissions, fetch the data.